docs/changes/io.fits/15501.feature.rst

```diff
@@ -0,0 +1,4 @@
+Add new option ``decompress_in_memory=False`` to ``fits.open``, to decompress
+the whole file in memory at once, instead of decompressing small blocks on the
```

Comment:
maybe you could say 'instead of decompressing the file progressively as data is needed' or similar, again avoiding the term block.